includes/class-wc-emails.php

```diff
@@ -269,6 +270,108 @@ public function order_details( $order, $sent_to_admin = false, $plain_text = fal
 	}
 
 	/**
+	 * Adds Schema.org markup for order in JSON-LD format.
+	 *
+	 * @since 2.6.0
+	 * @param mixed $order
+	 * @param bool $sent_to_admin (default: false)
+	 * @param bool $plain_text (default: false)
+	 */
+	public function order_schema_markup( $order, $sent_to_admin = false, $plain_text = false ) {
+		if ( $plain_text ) {
+			return;
+		}
+
+		$accepted_offers = array();
+
+		foreach ( $order->get_items() as $item ) {
+			if ( ! apply_filters( 'woocommerce_order_item_visible', true, $item ) ) {
+				continue;
+			}
+
+			$product = apply_filters( 'woocommerce_order_item_product', $order->get_product_from_item( $item ), $item );
+			$is_visible = $product && $product->is_visible();
+
+			$item_offered = array(
+				'@type' => 'Product',
+				'name' => apply_filters( 'woocommerce_order_item_name', $item['name'], $item, $is_visible )
+			);
+
+			if ( $sku = $product->get_sku() ) {
+				$item_offered['sku'] = $sku;
+			}
+
+			if ( $is_visible ) {
+				$item_offered['url'] = get_permalink( $product->get_id() );
+			}
+
+			if ( $image_id = $product->get_image_id() ) {
+				$item_offered['image'] = wp_get_attachment_image_url( $image_id, 'thumbnail' );
+			}
+
+			$accepted_offer = array(
+				'@type' => 'Offer',
+				'itemOffered' => $item_offered,
```

Comment:
All the new arrays introduced by your PR should have each value indented so they align. See https://make.wordpress.org/core/handbook/best-practices/coding-standards/php/#indentation for more information.